core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -6928,6 +6938,21 @@ private void checkNullOperand(SqlTester tester, String op) {
         "0",
         "BIGINT NOT NULL");
 
+    tester.checkScalar(
+        "extract(millisecond from date '2008-2-23')",
+        "0",
+        "BIGINT NOT NULL");
+
+    tester.checkScalar(
+        "extract(microsecond from date '2008-2-23')",
+        "0",
+        "BIGINT NOT NULL");
+
+    tester.checkScalar(
+        "extract(nanosecond from date '2008-2-23')",
+        "0",
+        "BIGINT NOT NULL");
+
```

Comment:
Can you add a case extracting millisecond/microsecond from a date before epoch?

Reply:
There is already a test case for extracting millisecond/microsecond from a date before epoch. See https://github.com/apache/calcite/pull/1331/files#diff-25687b6ff3e28fb79b62ba5050883d45R7290